packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceConfigSet.test.ts

```diff
@@ -5,35 +5,66 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
+import { Config } from '@salesforce/core';
+import { Table } from '@salesforce/salesforcedx-utils-vscode/out/src/output';
 import { expect } from 'chai';
-import { ForceConfigSetExecutor } from '../../../src/commands';
+import * as sinon from 'sinon';
+import { channelService } from '../../../src/channels';
+import { forceConfigSet, ForceConfigSetExecutor } from '../../../src/commands';
 import { nls } from '../../../src/messages';
 
+const CONFIG_KEY = nls.localize('force_config_set_name');
+const CONFIG_TITLE = nls.localize('force_config_set_title');
+
+const sandbox = sinon.createSandbox();
+let channelSpy: sinon.SinonSpy;
+let configSetSpy: sinon.SinonSpy;
+let configWriteSpy: sinon.SinonSpy;
+let tableSpy: sinon.SinonSpy;
+
 describe('Force Config Set', () => {
-  it('should build the force config set command', async () => {
+  beforeEach(() => {
+    channelSpy = sandbox.spy(channelService, 'appendLine');
+    configSetSpy = sandbox.spy(Config.prototype, 'set');
+    configWriteSpy = sandbox.spy(Config.prototype, 'write');
+    tableSpy = sandbox.spy(Table.prototype, 'createTable');
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('should set config with the given username or alias', async () => {
     const usernameOrAlias = 'test-username1@gmail.com';
-    const forceConfigSet = new ForceConfigSetExecutor(usernameOrAlias);
-    const forceConfigSetCommand = forceConfigSet.build({});
-    expect(forceConfigSetCommand.toCommand()).to.equal(
-      `sfdx force:config:set defaultusername=${usernameOrAlias}`
-    );
-    expect(forceConfigSetCommand.description).to.equal(
-      nls.localize('force_config_set_org_text')
-    );
+    await forceConfigSet(usernameOrAlias);
+    expect(configSetSpy.calledOnce);
+    expect(configSetSpy.calledWith(CONFIG_KEY, usernameOrAlias)).to.equal(true);
+    expect(configWriteSpy.calledOnce);
   });
-});
 
-describe('Force Config Set using multiple aliases for a single username', () => {
-  it('should build the force config set command with first alias', async () => {
+  it('should set config with first alias', async () => {
```

Comment:
Love to see this test covered here. Is there any scenario in which there would be no alias to set? It's more likely that we'd encounter this if the user was making modifications to that file directly, or through another program.

Reply:
I've been looking into that, and the only situation where the Set Config command gets called is when the user selects a valid alias from a dropdown. So invalid and blank inputs can not be entered.